binderhub/tests/conftest.py

```diff
@@ -13,23 +18,62 @@
 root = os.path.join(here, os.pardir, os.pardir)
 minikube_testing_config = os.path.join(root, 'testing', 'minikube', 'binderhub_config.py')
 
+TEST_NAMESPACE = os.environ.get('BINDER_TEST_NAMESPACE') or 'binder-test'
+KUBERNETES_AVAILABLE = False
+
+ON_TRAVIS = os.environ.get('TRAVIS')
+
+# set BINDER_TEST_URL to run tests against an already-running binderhub
+# this will skip launching BinderHub internally in the app fixture
+BINDER_URL = os.environ.get('BINDER_TEST_URL')
+REMOTE_BINDER = bool(BINDER_URL)
+
+
 @pytest.fixture(scope='session')
 def _binderhub_config():
+    """Load the binderhub configuration
+
+    Currently separate from the app fixture
+    so that it can have a different scope (only once per session).
+    """
     cfg = PyFileConfigLoader(minikube_testing_config).load_config()
-    cfg.BinderHub.build_namespace = 'binder-test'
+    cfg.BinderHub.build_namespace = TEST_NAMESPACE
+    if ON_TRAVIS:
+        cfg.BinderHub.hub_url = cfg.BinderHub.hub_url.replace('192.168.99.100', '127.0.0.1')
```

Comment:
https://github.com/jupyterhub/binderhub/blob/1951fe70957f7e1c4472e5cd64873d6ffb49dc15/testing/minikube/binderhub_config.py now asks minikube for the IP, on travis will we end up in the except clause of that or not?

Reply:
I think this will work most of the time and fail rarely so if we are in a hurry merge first, fix later.

Reply:
This line becomes obsolete thanks to [another PR](https://github.com/jupyterhub/binderhub/pull/266/files#diff-8f47f4b62ab8938d604ab0e0ffa506bd). I haven't rebased this PR since merging that one, so I will remove it in a subsequent one.